src/pymmcore_widgets/_util.py

```diff
@@ -114,71 +115,57 @@ def fov_kwargs(core: CMMCorePlus) -> dict:
     return {}
 
 
-def _get_next_available_path(
-    path: Path | str, extension: str, ndigits: int = 3
-) -> Path:
+# examples:
+# "name_001" -> ("name", "001")
+# "name" -> ("name", "")
+# "name_001_002" -> ("name_001", "002")
+# "name_02" -> ('name_02', None)
+NUM_SPLIT = re.compile(r"(.*?)(?:_(\d{3,}))?$")
+
+
+def get_next_available_path(requested_path: Path | str, min_digits: int = 3) -> Path:
     """Get the next available paths (filepath or folderpath if extension = "").
 
-    This method adds a counter of ndigits to the filename or foldername to ensure
+    This method adds a counter of min_digits to the filename or foldername to ensure
     that the path is unique.
 
     Parameters
     ----------
-    path : Path | str
-        The starting path without extension (e.g./User/Desktop/Folder/Filename).
-    extension : str
-        The extension to be used (e.g. ".ome.tiff").
-    ndigits : int (optional)
-        The number of digits to be used for the counter. By default, 3.
+    requested_path : Path | str
+        A path to a file or folder that may or may not exist.
+    min_digits : int, optional
+        The min_digits number of digits to be used for the counter. By default, 3.
     """
-    # if the extension does not start with a dot, add it
-    if extension and not extension.startswith("."):
-        extension = f".{extension}"
-
-    if isinstance(path, str):
-        path = Path(path)
-
-    # remove extension from the path if any
-    if str(path).endswith(extension):
-        path = Path(str(path).replace(extension, ""))
-
-    stem = path.stem
-
-    # remove digits from the stem if any
-    cur_num = stem.rsplit("_")[-1]
-    if cur_num.isdigit() and len(cur_num) == ndigits:
-        stem = stem[: -ndigits - 1]
-        # if the stem provided has a number, check if path exists and if not, return it
-        new_path = path.parent / f"{stem}_{cur_num}{extension}"
-        if not new_path.exists():
-            return new_path
```

Comment:
I think this is the only behavior that is missing but I guess is not important and probably never happens in a normal use.

Basically, if in a `folder` you have a file called `test_001.ome.tif` and the `path` arg in this function is `folder/test_005.ome.tif`, the returned path is `folder/test_005.ome.tif` and not `folder/test_002.ome.tif`.


This is a simple test if you want to see the behavior:

```py
def test_path(tmp_path: Path) -> None:
    (tmp_path / "test_001.ome.tif").touch()
    path = get_next_available_path(tmp_path / "test_005.ome.tif")
    assert path.name == "test_005.ome.tif"
```

Reply:
ah yep, good catch.  provided that `005` is greater than current_max, i agree.  will fix